_posts/2018-02-15-upgrading-a-monolith.markdown

```diff
@@ -0,0 +1,19 @@
+---
+layout: post
+title: "Upgrading a monolith from Rails 4.0 to 5.1"
+date: 2018-02-15
+categories: ["rails", "upgrades", "case-study"]
+author: "emily"
+---
+
+We recently completed work with [Power Home Remodeling](https://powerhrg.com) on a Rails upgrade for their self-described “monolith CRM/ERP application”, and were able to speak to them about their experience with Ombu Labs.
+
+<!--more-->
+
+We talked to [Ben Langfeld](https://www.linkedin.com/in/benlangfeld/), the Application Solution Architect at [Power Home Remodeling](https://powerhrg.com), about the work on their app, Nitro. According to Ben, their monolithic CRM/ERP application is continuously built by a team of approximately 50 developers, system administrators, testers and support staff. The application contains *over 500,000 lines of Ruby on Rails and Javascript code*, and by their own definition, is one of the more complex Rails applications out there in terms of scope.
+
+Like many companies, [Power Home Remodeling](https://powerhrg.com) was having difficulty allocating developer attention to the Rails upgrade project because of the demands of feature work coming from other areas of the business. That is why they decided to come to Ombu Labs for help.
+
+We executed a **full upgrade of the application from Rails 4.0 to 5.1** and prepared the company for an eventual 5.2 upgrade. This necessary upgrade of their application led to a “reduction of risk in terms of repeat bugfix/feature development work compared to what is available in a more modern stack.”
```

Comment:
@ercohen14 Could you add a link to the `5.2` -> http://weblog.rubyonrails.org/2017/11/27/Rails-5-2-Active-Storage-Redis-Cache-Store-HTTP2-Early-Hints-Credentials/